src/PointwiseFunctions/Hydro/EquationsOfState/HybridEos.cpp

```diff
@@ -0,0 +1,153 @@
+    const Scalar<DataType>& rest_mass_density,
+    const Scalar<DataType>& specific_enthalpy) const noexcept {
+  if constexpr (ColdEquationOfState::is_relativistic) {
+    return Scalar<DataType>{
+        (get(cold_eos_.pressure_from_density(rest_mass_density)) +
+         get(rest_mass_density) * (thermal_adiabatic_index_ - 1.0) *
+             (get(specific_enthalpy) - 1.0 -
+              get(cold_eos_.specific_internal_energy_from_density(
+                  rest_mass_density)))) /
+        thermal_adiabatic_index_};
+  } else {
+    return Scalar<DataType>{
+        (get(cold_eos_.pressure_from_density(rest_mass_density)) +
+         get(rest_mass_density) * (thermal_adiabatic_index_ - 1.0) *
+             (get(specific_enthalpy) -
+              get(cold_eos_.specific_internal_energy_from_density(
+                  rest_mass_density)))) /
+        thermal_adiabatic_index_};
+  }
+}
+
+template <typename ColdEquationOfState>
+template <class DataType>
+Scalar<DataType>
+HybridEos<ColdEquationOfState>::specific_enthalpy_from_density_and_energy_impl(
+    const Scalar<DataType>& rest_mass_density,
+    const Scalar<DataType>& specific_internal_energy) const noexcept {
+  return Scalar<DataType>{
+      get(cold_eos_.specific_enthalpy_from_density(rest_mass_density)) +
+      thermal_adiabatic_index_ *
+          (get(specific_internal_energy) -
+           get(cold_eos_.specific_internal_energy_from_density(
+               rest_mass_density)))};
+}
+
+template <typename ColdEquationOfState>
+template <class DataType>
+Scalar<DataType> HybridEos<ColdEquationOfState>::
+    specific_internal_energy_from_density_and_pressure_impl(
+        const Scalar<DataType>& rest_mass_density,
+        const Scalar<DataType>& pressure) const noexcept {
+  return Scalar<DataType>{
+      get(cold_eos_.specific_internal_energy_from_density(rest_mass_density)) +
+      1.0 / (thermal_adiabatic_index_ - 1.0) *
+          (get(pressure) -
+           get(cold_eos_.pressure_from_density(rest_mass_density))) /
+          get(rest_mass_density)};
+}
+
+template <typename ColdEquationOfState>
+template <class DataType>
+Scalar<DataType>
+HybridEos<ColdEquationOfState>::chi_from_density_and_energy_impl(
+    const Scalar<DataType>& rest_mass_density,
+    const Scalar<DataType>& specific_internal_energy) const noexcept {
+  return Scalar<DataType>{
+      get(cold_eos_.chi_from_density(rest_mass_density)) +
+      (thermal_adiabatic_index_ - 1.0) *
+          (get(specific_internal_energy) -
+           get(cold_eos_.specific_internal_energy_from_density(
+               rest_mass_density)) -
+           get(cold_eos_.pressure_from_density(rest_mass_density)) /
+               get(rest_mass_density))};
+}
+
+template <typename ColdEquationOfState>
+template <class DataType>
+Scalar<DataType> HybridEos<ColdEquationOfState>::
+    kappa_times_p_over_rho_squared_from_density_and_energy_impl(
+        const Scalar<DataType>& rest_mass_density,
```

Comment:
How is this define in Spectre? I am used to kappa being the partial derivative of the pressure with respect to the specific internal energy at constant density, which is just (thermal_adiabatic_index-1); but this seems more complex.

Reply:
Ah, right, missed the (x P); that's fine then, everything after the (Gamma-1) is just P/rho expressed as a function of rho,epsilon...

Reply:
ah, okay

Reply:
The current version in the polytropic equation of state does lead to the correct value of the sound speed, but it probably won't be 'chi' when extended to T!=0

Reply:
the function definition for `chi` is right before the definition for `kappa` and adds a correction to the cold value

Reply:
Got it - sorry to have missed that function, and I agree with the expression there. 
One potentially confusing part is that the correction term is not zero when T=0, as 
chi=dP/drho(T=0)
for the 1D EoS, but
chi=\partial P/\partial rho 
at constant epsilon for 2D EoS; so the first is a derivative at constant T and the second at constant epsilon. 
Maybe writing dP/drho instead of partial derivatives on L136 of EquationOfState.hpp would help?